low-level/files/opt/seagate/sspl/bin/generate_sspl_bundle/sspl_bundle_generate

```diff
@@ -110,125 +168,154 @@ def make_sspl_support_bundle_tarfile(parser, tmpFile):
+        consuldump_cmd = {
+            "sspl_config": f"{consuldump} consul_sspl_config sspl/",
+            "sspl_cache": f"{consuldump} consul_sspl_cache var/{PRODUCT_FAMILY}/sspl",
+            "system_information": f"{consuldump} consul_system_information system_information/",
+            "storage_enclosure": f"{consuldump} consul_storage_enclosure storage_enclosure/",
+            "rabbitmq": f"{consuldump} consul_rabbitmq rabbitmq/"
+        }
+        for key, value in consuldump_cmd.items():
+            subout = self._run_command(value)
+            if subout[2] != 0:
+                print(f"Unable to get consul data of {key}...!")
+            del subout
+        consul_folder_path = f"/tmp/{sspl_bundle_tmp}/{self.consul_folder_name}"
+        self.consul_sspl_config = f"{consul_folder_path}/consul_sspl_config"
+        self.consul_sspl_cache = f"{consul_folder_path}/consul_sspl_cache"
+        self.consul_system_information = f"{consul_folder_path}/consul_system_information"
+        self.consul_storage_enclosure = f"{consul_folder_path}/consul_storage_enclosure"
+        self.consul_rabbitmq = f"{consul_folder_path}/consul_rabbitmq"
+
+    def get_drives_smart_data(self):
+        lsscsi_cmd = "lsscsi | grep disk"
+        lsscsi_response, _, return_code = self._run_command(lsscsi_cmd)
+        if not os.path.exists(self.boot_drvs_dta):
+            os.makedirs(self.boot_drvs_dta)
+        for res in lsscsi_response.split("\n"):
+            drive_path = res.strip().split(' ')[-1]
+            smartctl_cmd = f"sudo smartctl -a {drive_path} --json"
+            response, _, return_code = self._run_command(smartctl_cmd)
+            response = json.loads(response)
+            # Filtering out local disk
+            if response['device']['protocol'] == 'ATA':
+                try:
+                    file_name = drive_path.replace('/', '_')[1:]
+                    with open(f"{self.boot_drvs_dta}/{file_name}.json", "w+") as fp:
+                        json.dump(response, fp,  indent=4)
+
+                except Exception as e:
+                    print("Error in writing {0} file: {1}".format(response, e))
+    
+    def _run_command(self, command):
+        """Run the command and get the response and error returned"""
+        process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
+        return_code = process.wait()
+        response, error = process.communicate()
+        return response.rstrip('\n'), error.rstrip('\n'), return_code
+
+    def get_enclosure_logs(self):
+        url = singleton_realstorencl.build_url("/")
+        COLLECTING_DEBUG_LOG_STARTED = False
+        for encl_trigger_log_retry_index in range(0, ENCL_TRIGGER_LOG_MAX_RETRY):
+            response = singleton_realstorencl.ws_request(url, singleton_realstorencl.ws.HTTP_POST, \
+                post_data=f'/api/collectDebugData/"{SUPPORT_REQUESTOR_NAME}"{SUPPORT_EMAIL_ID}"{SUPPORT_CONTACT_NUMBER}')
```

Comment:
api uri was always hardcoded here, can we move this to realstor_enclosure.py & use it from there, like we do for other apis. Intention is to keep all use APIs list in one place.

Reply:
Done

Reply:
@satish-seagate why following uri is still seen above ?
> '/api/collectDebugData/

Reply:
Done